### ISSUE TITLE
Is Yue deprecated?

### ISSUE THREAD

**Issue by ghost** [yue/yue#90]
The main Yue repo has not had a commit for 5 months. I am wondering if work of Yue has been halted, and if so, for how long?

**Comment by zcbenz** [comment]
It is not deprecated but the development depends on my free time and passion for open source, which I don't have probably until next year.

**Comment by cancerberoSgx** [comment]
This library is awesome and unique. I've been looking to have a native lightweight library for node.js (or something even in C that I could port / bind) and this is my third time looking for and fortunately I found this working fine in linux and mac and node.js support with typings! Thanks , keep it up, really this is unique 

@brushtool I consider a library active and well maintained if last commit was 5 months ago...

**Comment by mdings** [comment]
@zcbenz I appreciate your commitment to this library. And to be honest, I think this is one of the best libs out there that both performs well and still is lightweight and flexible/customizable. That's a truly unique offering and IMHO that's something people are looking for nowadays (just google for "Electron package size") 😀 I think it's just a matter of gaining attention and traction before this lib wil be even more popular.

However, with the last commit being > year and no comments in the issues, it feels like Yue is being deprecated or abandoned. Please, don't get me wrong. I understand where you're coming from and that maintaining open-source libs with everything around it, can consume lots of time and efforts. I've been no stranger to offering "abandonware" myself in the past unfortunately.. And I'm not proud of it.

Still, I think it would be nice if you could share your vision and or roadmap around Yue. Are you still planning on committing to the further development? And if so, is there something the community can assist with? Moreover, you might find it beneficial to setup a patreon - or similar - account, so people can financially support you as well. If not, please let everyone know so we can choose alternatives deliberately.

**Comment by zcbenz** [comment]
@mdings Thank you for your kind words, it does give me more power to work on this project.

Currently the main bottleneck of the development, is Windows support. Yue uses Win32 APIs and GDI+, which are decades old and it is extremely hard to implement a modern UI on them. And as Microsoft is deprecating Windows 7, it makes less sense to use the old techs.

I plan to use to [WinUI 3.0](https://github.com/microsoft/microsoft-ui-xaml) to rewrite Windows support, which is modern and officially supported by Microsoft. However it is still being developed and integrating it to Yue would be a rather difficult problem. And adopting it also means Yue would be Windows 10 only.

I'll find some time later to do some simple maintenance work (bug fixes, simple features, language binding updates, roadmaps), but major new features (for example more rendering APIs, shinny new controls) will still be halted until the I rewrite the Windows support with WinUI, and I would expect it be 1 or 2 years later.

**Comment by gamelaster** [comment]
I'm watching this project for couple of years, I totally agree with mdings's comment upper, just one note from me. By my opinion, deprecating GDI+ is bad idea by my opinion, personally, I loved that Yue is small, performing, lightweight, and can run on multiple platforms, including older Windows.
Wouldn't be better to use for example, DirectX 9 or Direct2D for Windows, so it will be easier to work with, but also it will be supported by older Windows releases?

Thanks for your work zcbenz :)
Regards

**Comment by zcbenz** [comment]
One factor of this project, is to draw controls with their native appearance, on Windows it is achieved with GDI+ and the Uxtheme APIs. However switching to Direct2D means we have to draw most controls ourselves, which is doable and most UI mature UI libraries are taking this approach, but this would be a great burden to maintain, and I would rather spend the time on moving to WinUI.

But anyway current Windows implementation will not be abandoned immediately, I'll continue to extend the Yue APIs with Win32 APIs in the next few years, probably by the time Yue has been moved to WinUI, the world may have already been moved away from Windows 7 completely.

**Comment by rubyFeedback** [comment]
There is definitely a use case - see libui, including language bindings.

On Linux I can use gtk + ruby-gtk just fine and it is working as well. On
windows this is all so annoyingly more complicated ... even the gtk guide
doesn't work reliably so. I got a missing symbol error when using GCC
there (I can compile simple .c programs though). I gave up afterwards
because I am chasing-the-stick here by investing time and it's still not
working on windows. Probably others have a similar use case - for me
it's actually getting small things work for elderly relatives who aren't
that tech-savvy. So these things must be simple from A to Z.

Of course the "use the www stack" is the more realistic variant to
go, which I have to do anyway I suppose - but I so much dislike
javascript ... would be really cool if we'd have C/C++ toolkits
and language bindings (pygobject of gtk, with all its speed 
issues, is still very nice - I'd love if toolkits could have something
like that. It's probably only realistic for toolkits that are lightweight
and stable/documented.)

Props for your heroic effort to battle the windows monster here - 
I'd never have enough patience to create and maintain a UI 
library per se, creating it from de-novo. (I have a gem for ruby-gtk
but here I depend on on upstream GTK and the bindings to
it provided by other folks. My brain is too feeble to work with
the raw metal underneath, I need simple DSLs and useful
abstractions.)
